.github/workflows/build.yml

```diff
@@ -0,0 +1,134 @@
+name: "Test & Build"
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up MongoDB
+        uses: ankane/setup-mongodb@v1
+        with:
+          mongodb-version: '5.0'
+
+      - name: Set up JDK 17 (Temurin)
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Compile
+        run: |
+          mvn -q -U -B compile
+
+      - name: Test
+        run: |
+          mvn -q -U -B test
+
+      - name: Package
+        run: |
+          mvn -q -U -B -ff -DskipTests package
+
+      - name: Verify
+        run: |
+          mvn -q -U -B -ff -DskipTests package
+
+  docker:
+    name: Docker build
+    runs-on: ubuntu-latest
+    needs: test
+
+    env:
+      PUBLIC_IMAGE: fairdata/fairdatapoint
+      PRIVATE_IMAGE: ${{ secrets.PRIVATE_REGISTRY_URL }}/fairdatapoint
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Check available platforms
+        run: echo ${{ steps.buildx.outputs.platforms }}
+
+      # PRIVATE: DOCKER REGISTRY
+      - name: Docker login  [private]
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ secrets.PRIVATE_REGISTRY_URL }}
+          username: ${{ secrets.PRIVATE_REGISTRY_USERNAME }}
+          password: ${{ secrets.PRIVATE_REGISTRY_PASSWORD }}
+
+      - name: Docker meta [private]
+        id: meta-private
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ env.PRIVATE_IMAGE }}
+          tags: |
+            type=ref,event=branch
+            type=semver,pattern={{version}}
```

Comment:
Did we always push versions to the private registry? I thought these were only in the public one.